wicket-request/src/main/java/org/apache/wicket/request/mapper/parameter/PageParameters.java

```diff
@@ -294,35 +293,48 @@ public PageParameters add(final String name, final Object value, final int index
 		Args.notEmpty(name, "name");
 		Args.notNull(value, "value");
 
-		if (namedParameters == null)
-		{
-			namedParameters = new ArrayList<>(1);
-		}
-
-		List<String> values = new ArrayList<>();
 		if (value instanceof String[])
 		{
-			values.addAll(Arrays.asList((String[])value));
+			addNamed(name, (String[]) value, index, type);
 		}
 		else
 		{
-			values.add(value.toString());
+			addNamed(name, value.toString(), index, type);
+		}
+
+		return this;
+	}
+
+	private void addNamed(String name, String[] values, int index, Type type) 
+	{
+		if (namedParameters == null)
```

Comment:
It would be better to add `&& values.length > 0`

Reply:
Done